examples/advanced_operations/add_responsive_search_ad_full.py

```diff
@@ -0,0 +1,535 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This example shows how to create a complete Responsive Search ad.
+
+Includes creation of: budget, campaign, ad group, ad group ad,
+keywords, geo targeting, and image extensions.
+
+More details on Responsive Search ads can be found here:
+https://support.google.com/google-ads/answer/7684791
+"""
+
+import argparse
+import sys
+import uuid
+import requests
+
+from google.ads.googleads.client import GoogleAdsClient
+from google.ads.googleads.errors import GoogleAdsException
+
+
+# Keywords from user.
+_KEYWORD_TEXT_EXACT_1 = "example of exact match"
+_KEYWORD_TEXT_PHRASE_1 = "example of phrase match"
+_KEYWORD_TEXT_BROAD_1 = "example of broad match"
+
+# Geo targeting from user.
+_GEO_LOCATION_1 = "Buenos aires"
+_GEO_LOCATION_2 = "San Isidro"
+_GEO_LOCATION_3 = "Mar del Plata"
+
+# _LOCALE and _COUNTRY_CODE are used for geo targeting.
+# _LOCALE is using ISO 639-1 format. If an invalid _LOCALE is given,
+# 'es' is used by default.
+_LOCALE = "es"
+
+# A list of country codes can be referenced here:
+# https://developers.google.com/google-ads/api/reference/data/geotargets
+_COUNTRY_CODE = "AR"
+
+
+def create_ad_text_asset(client, text, pinned_field=None):
+    """Create an AdTextAsset.
+    Args:
+        client: an initialized GoogleAdsClient instance.
+        text: text for headlines and descriptions.
+        pinned_field: to pin a text asset so it always shows in the ad.
+    
+    Returns:
+        An ad text asset.
+    """
+    ad_text_asset = client.get_type("AdTextAsset")
+    ad_text_asset.text = text
+    if pinned_field:
+        ad_text_asset.pinned_field = pinned_field
+    return ad_text_asset
+
+
+def main(client, customer_id, omit_image_extensions):
+    """
+    The main method that creates all necessary entities for the example.
+
+    Args:
+        client: an initialized GoogleAdsClient instance.
+        customer_id: a client customer ID.
+
+    Returns:
+        A responsive search ad with all settings required to run campaign.
+    """
+    # Create a budget, which can be shared by multiple campaigns.
+    campaign_budget = create_campaign_budget(client, customer_id)
+
+    campaign_resource_name = create_campaign(client, customer_id, campaign_budget)
+
+    ad_group_resource_name = create_ad_group(client, customer_id, campaign_resource_name)
+
+    create_ad_group_ad(client, customer_id, ad_group_resource_name)
+
+    add_keywords(client, customer_id, ad_group_resource_name)
+
+    add_geo_targeting(client, customer_id, campaign_resource_name)
+
+    # This is optional but recommended for RSA.
+    # To add image extensions, the account has to follow these requirements:
+    # https://support.google.com/google-ads/answer/9566341
+    # If the account meets the requirements, set below variable to True.
+    if omit_image_extensions:
```

Comment:
This should be `if not omit_image_extensions`, correct?

On that note, could we make this `include_image_extensions` and have it default to `True` instead? That would make the logic more readable and avoid a double negative for the "do something" case.

Reply:
You are right! Let's change it to include_image_extensions, but I would still have it as False as a default because not all accounts can create them. Therefore, we avoid developers getting errors.